src/integrationTest/java/org/opensearch/security/FlsAndFieldMaskingTests.java

```diff
@@ -833,6 +906,789 @@ public void getFieldCapabilities() throws IOException {
     }
 
     @Test
+    public void testGetDocumentWithNoTitleFieldOrOnlyTitleFieldFLSRestrictions() throws IOException, Exception {
+        GetRequest getRequest = new GetRequest(FIRST_INDEX_NAME, FIRST_INDEX_ID_SONG_1);
+
+        try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(USER_ONLY_FIELD_TITLE_FLS)) {
+            assertProperGetResponsesForCorrespondingFLSRestrictions(restHighLevelClient, getRequest, true);
+        }
+
+        try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(USER_NO_FIELD_TITLE_FLS)) {
+            assertProperGetResponsesForCorrespondingFLSRestrictions(restHighLevelClient, getRequest, false);
+        }
+    }
+
+    private void assertProperGetResponsesForCorrespondingFLSRestrictions(
+        RestHighLevelClient restHighLevelClient,
+        GetRequest getRequest,
+        boolean getOne
```

Comment:
nit: can be renamed to `shouldShowFieldTitle`

Reply:
also, method names can be shortened to: `assertGetForFLSRestrictions`, `assertMGetsForFLSRestrictions`, `assertSearchForFLSRestrictions`, etc.